src/app/components/notification-box/notification-box.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit } from '@angular/core';
+import { timer, interval, Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { NotificationSettingsService } from '../../services/notification-settings.service';
+import { NotificationSettings } from '../../models/notification-settings-models/notification-settings';
+
+@Component({
+    selector: 'app-notification-box',
+    templateUrl: './notification-box.component.html',
+    styleUrls: ['./notification-box.component.scss']
+})
+export class NotificationBoxComponent implements OnInit {
+    notificationSettings: NotificationSettings;
+    timer: Observable<number>;
+    timerSubscription: Subscription;
+    intervalSubscription: Subscription;
+    viewsCounter = 0;
+    ticks: number = 1000;
+    progress: number = 0;
+    isVisible = false;
+    interval = 10;
+
+    constructor(
+        private notifactionSettingsService: NotificationSettingsService
+    ){}
+
+    ngOnInit() {
+        this.notificationSettings = this.notifactionSettingsService.getConfiguration();
+        this.startTimer();
+    }
+
+    startTimer() {
+        this.timer = timer(this.notificationSettings.delayTime, 1000);
+
+        this.timerSubscription = this.timer.subscribe(() => {
```

Comment:
ngDestroy may have "this.timerSubscription.unsubscribe()"

Reply:
Fixed